src/services/reviews.js

```diff
@@ -0,0 +1,672 @@
+	 * @name verifyAndFetchReview
+	 * @param {String} userId - user id
+	 * @param {Integer} resourceId - resource id
+	 * @returns {JSON} - return error if any
+	 */
+
+	static async verifyAndFetchReview(userId, resourceId) {
+		//Check if the logged-in user started the review
+		const reviewResource = await reviewResourceQueries.findOne({
+			reviewer_id: userId,
+			resource_id: resourceId,
+		})
+
+		// If the review resource is not found, return an error response
+		if (!reviewResource || !reviewResource.id) {
+			return responses.failureResponse({
+				message: 'REVIEW_NOT_FOUND',
+				statusCode: httpStatusCode.bad_request,
+				responseCode: 'CLIENT_ERROR',
+			})
+		}
+
+		// Fetch the review
+		const review = await reviewsQueries.findOne({
+			organization_id: reviewResource.organization_id,
+			resource_id: resourceId,
+			reviewer_id: userId,
+		})
+
+		// If the review is not found, return an error response
+		if (!review || !review.id) {
+			return responses.failureResponse({
+				message: 'REVIEW_NOT_FOUND',
+				statusCode: httpStatusCode.bad_request,
+				responseCode: 'CLIENT_ERROR',
+			})
+		}
+
+		return responses.successResponse({
+			statusCode: httpStatusCode.ok,
+			result: review,
+		})
+	}
+
+	/**
+	 * Check the review is already started by someone
+	 * @method
+	 * @name validateNoActiveReviewByOthers
+	 * @param {String} userId - user Id
+	 * @param {Integer} resourceId - resource Id
+	 * @param {String} orgId - organization Id
+	 * @returns {JSON} - return error if any
+	 */
+
+	static async validateNoActiveReviewByOthers(resourceId, userId, orgId) {
+		// Check if there's an existing active review by another user
+		const existingReview = await reviewsQueries.findOne({
+			organization_id: orgId,
+			resource_id: resourceId,
+			reviewer_id: { [Op.not]: userId },
+			status: { [Op.in]: _restrictedReviewStatuses },
+		})
+
+		// If an active review exists by another user, return an error response
+		if (existingReview?.id) {
+			return responses.failureResponse({
+				message: 'REVIEW_INPROGRESS',
+				statusCode: httpStatusCode.bad_request,
+				responseCode: 'CLIENT_ERROR',
+			})
+		}
+
+		// If no active review is found, return a success response
+		return responses.successResponse({
+			statusCode: httpStatusCode.ok,
+		})
+	}
+
+	/**
+	 * Validate the reviewer level matches the resource level
+	 * @method
+	 * @name validateNextLevel
+	 * @param {Array<Object>} roles - user roles
+	 * @param {String} userOrgId - organization Id
+	 * @param {String} resourceType - Resource type
+	 * @param {Integer} currentLevel - current stage of resource
+	 * @returns {JSON} - return error if any
+	 */
+	static async validateNextLevel(roles, userOrgId, resourceType, currentLevel) {
+		//get the roles
+		const userRoleTitles = utils.getUniqueElements(roles.map((item) => item.title))
+		//validate the level
+		const resourceWiseLevels = await resourceService.fetchReviewLevels(userOrgId, userRoleTitles, [resourceType])
```

Comment:
@priyanka-TL Add sample arguments, better descripton and sample response of this function. Try to come up with a better name.